### PR TITLE
refactor(fixer): add span to message

### PR DESCRIPTION
**Pull request by jason89521**
## Description

- Move `autofix/` to `fixer.rs`.
- Add span to `Message` such that we can pass the message sorting test case.
- Remove `fix` method from `LintContext`.

### CODE REVIEW

**Comment by jason89521** [comment]
The `labels` method return an iterator of label, how should we compare the two messages by this method?

**Comment by Boshen** [comment]
> The `labels` method return an iterator of `LabelSpan`s, how should we compare the two messages by this method?

Find the minimum span.start I think. You have to get used to span being unique and sorted ;-)

**Comment by jason89521** [comment]
Does our rule at least contain one span?

**Comment by Boshen** [comment]
> 

Yes, you can assume and use `expect` here.

**Comment by jason89521** [comment]
The `LabeledSpan` doesn't has `start`, so I use `offset` and `len` to sort the messages.

File: crates/oxc_linter/src/fixer.rs [hunk 1]
```diff
@@ -92,7 +91,25 @@ impl<'a> Fixer<'a> {
         output.push_str(&source_text[offset..]);
 
         let mut messages = self.messages.into_iter().filter(|m| !m.fixed).collect::<Vec<_>>();
-        messages.sort_by_key(|m| m.span);
+        messages.sort_by(|a, b| {
```
Comment (Boshen) [inline crates/oxc_linter/src/fixer.rs:94]: Can you take a look at `sort_by_cached_key` and `sort_unstable_by_key`

Comment (jason89521) [reply crates/oxc_linter/src/fixer.rs:94]: But our sorting is determined by two key: `offset()` and `len()`, can we use one of the above methods?

Comment (jason89521) [reply crates/oxc_linter/src/fixer.rs:94]: Oh we can use tuple here

**Review by Boshen — APPROVED** [review]

**Pull request merged**